ecmd-core/ext/fapi2/capi/target.H

```diff
@@ -303,6 +304,49 @@ Target<K, M, V>::getChildren(const TargetFilter i_filter,
     return l_children;
 }
 
+template<TargetType K, MulticastType M, typename V>
+template< TargetType T >
+inline ReturnCodes
+Target<K, M, V>::reduceType(Target<T, M, V>& o_target) const
+{
+    check_reduceType<T>();
+
+    bool compatible = true;
+    plat_target_handle_t l_handle = this->get();
+    const ecmdChipTarget * l_pEcmdTarget = l_handle;
+    fapi2::TargetType l_type = TARGET_TYPE_NONE;
+
+    uint32_t l_rc = fapi2GetTargetType(*l_pEcmdTarget, l_type);
+    if (l_rc)
+    {
+        compatible = false;
+    }
+
+    // Here you would check that the platform target's type is compatible with T,
+    // also taking multicast into account (iff the plat target is a real multicast
+    // target, T must include TARGET_TYPE_MULTICAST).
+
+    if (((T & TARGET_TYPE_MULTICAST) == TARGET_TYPE_NONE) &&
+        (l_handle.multicast == true))
+    {
+        compatible = false;
+    }
+
+    if (((l_type & T) == TARGET_TYPE_NONE) &&
+        (l_handle.multicast == false))
```

Comment:
The target type must always match regardless of multicast-ness. Apologies if the comment above wasn't clear enough.

Reply:
When the target is multicast, the underlying target type (l_type) will be a PROC_CHIP and will not match any of the types. Therfore `l_type` is only valid to check when it is not an actual multicast target.

Any checking that can be done statically to verify matching types when multicast should be done with check_reduceType()

Reply:
Ohhh, I see! In the SBE we store the underlying target type along with multicast targets so we can tell say core-multicast targets apart from eq-multicast, and we can leverage that information for this check. If you don't have the information you can't check it, agree :)